File: src/main/java/org/paul/YumlFormatter.java
package org.paul;

import java.util.List;

public class YumlFormatter implements UmlFormatter {

    public enum Mode { SIMPLE, CLASSES }

    private final Mode mode;

    public YumlFormatter(Mode mode) {
        this.mode = mode;
    }

    @Override
    public String format(List<Class<?>> classes) {
        return "";
    }
}

File: csharp/CSharpDecompiler/Formatter/YumlFormatter.cs
using CSharpDecompiler.Config;
using CSharpDecompiler.Model;

namespace CSharpDecompiler.Formatter;

public sealed class YumlFormatter(YumlFormatter.Mode mode) : IUmlFormatter
{
    public string Format(IReadOnlyList<TypeInfo> types, DecompileConfig config)
    {
        throw new NotImplementedException();
    }

    public enum Mode
    {
        Simple,
        Classes,
    }
}
